feat(core): Automatically extract evaluation metrics (no-changelog)

OlegIvaniv opened the pull request:
## Summary

This PR implements automatic metrics extraction from evaluation workflow result data. Because it's using the new Evaluation Metrics node, #14050 needs to be merged first.

## Key Changes
1. Removed Separate Controller
	- The metrics controller (metrics.controller.ts) has been removed
	- Metrics are now automatically extracted from the evaluation workflow rather than being managed separately
2. Simplified Metrics Extraction
	- Instead of using the last node output, metrics are now collected from all EvaluationMetrics nodes in the workflow
	- The system merges metrics from multiple nodes into a single result object
3. Removed UI for Manual Metrics Management
	- The frontend UI for managing metrics has been removed (MetricsInput.vue)
	- Metrics sections have been removed from the test definition form
4. Removed Error Cases
	- The UNKNOWN_METRICS error code has been removed as metrics are now automatically synced

## How It Works Now

1. Defining Metrics
	- Users add one or more EvaluationMetrics nodes to their evaluation workflow
	- Each node defines metrics using name/value pairs, with values restricted to numbers
2. Running Tests
	- When a test is run, the system first syncs metrics by scanning the evaluation workflow
	- It finds all EvaluationMetrics nodes and extracts metric names
	- It updates the database by adding new metrics and removing unused ones
3. Collecting Results
	- When each test case is executed, results are collected from all EvaluationMetrics nodes
	- Results from multiple nodes are merged into a single metrics object
	- This object is then validated and stored with the test case execution

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

OlegIvaniv committed:
feat: Add Evaluation Metrics node and support for default assignment type

OlegIvaniv committed:
feat: Extract metrics from evaluation workflow

OlegIvaniv committed:
Remove metrics from front-end

OlegIvaniv committed:
Fix unit tests

OlegIvaniv committed:
Add basic tests

OlegIvaniv committed:
Make metrics plural

OlegIvaniv committed:
Remove missing metrics validation

OlegIvaniv committed:
Update evaluation workflow template

OlegIvaniv committed:
Linting fixes

OlegIvaniv committed:
Use constant for EVALUATION_METRICS_NODE

jeanpaul commented on packages/cli/src/evaluation.ee/test-runner/__tests__/evaluation-metrics.ee.test.ts:
Should we add (a) test(s) for multiple evaluation metrics nodes? That it picks up the different names across the nodes and/or takes the last executed one if multiple nodes specify the same metric?

jeanpaul commented on packages/cli/src/evaluation.ee/test-runner/test-runner.service.ee.ts:
```diff
@@ -359,6 +421,9 @@ export class TestRunnerService {
+			await this.syncMetrics(test.id, evaluationWorkflow);
+
 			// Get the metrics to collect from the evaluation workflow
 			const testMetricNames = await this.getTestMetricNames(test.id);
```
This is now doing the same query as in `syncMetrics` -- perhaps better would be to return the metric names from `syncMetrics` in order to avoid an extra DB query?

jeanpaul reviewed (changes requested):
Looks good already! Couple of small comments that would be good to check out

OlegIvaniv committed:
Merge branch 'master' into ai-769-add-metrics-node_rebased

OlegIvaniv committed:
Add string to number conversion and error handling for invalid values

OlegIvaniv committed:
Enhance defaultType handling in assignments

- Updated defaultType prop to be undefined by default.
- Added tests to verify defaultType behavior for new assignments, including drag-and-drop scenarios.
- Ensured that defaultType is respected when disableType is true and for all assignments when specified.

OlegIvaniv committed:
Merge branch 'ai-769-add-metrics-node_rebased' into ai-773-retrieve-metrics-definition-from-new-metric-nodes

OlegIvaniv committed:
Add tests for handling metrics in the middle of workflow, fix override order

OlegIvaniv committed:
Return metrics from syncMetrics and remove getTestMetricNames

OlegIvaniv committed:
Merge branch 'master' into ai-773-retrieve-metrics-definition-from-new-metric-nodes

janober commented:
Got released with [`n8n@1.86.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.86.0)
